### ISSUE TITLE
Troubleshooting running a Sonobus/aoo-compatible aooserver

### ISSUE THREAD

**Issue by paulreimer** [essej/aooserver#4]
I am attempting to run an additional aooserver to aoo.sonobus.net, mostly as a learning exercise but also to further develop various AOO clients -- and hoping to keep everything compatible with all existing Sonobus clients.

I would like to know the `aooserver` version/git commit currently running on aoo.sonobus.net; I am able to use Sonobus 1.6.1 with a locally compiled `aooserver` from the `master` branch as well as `aoonew`, but not `aoonewest`/`aoonewester`.

In an AOO client I hacked together (iOS), I am able to use aoo.sonobus.net, but for some reason it won't work with any `aooserver` that I have compiled myself -- I get `client_endpoint: connection was closed` errors; still trying to investigate more about this. But I'm hoping to start from an identical `aooserver` to what is running on aoo.sonobus.net, since that does work very well.

**Comment by essej** [comment]
Correct, do not attempt to use aoonewest or aoonewester, that is future research and development using a new version of the AOO lib, and the protocol has changed there... just stick to the develop branch (or main).

**Comment by actionless** [comment]
i've built it from master branch but same result - it starts and clients connect to it, but nothing happens,

only from time to time it writes ` aoo_server: not an AOO NET message!`

with (aoo.sonobus.net it also works fine, same as for original issue reporter)

also when using direct connection

UPD: my bad - i was trying to connect specifying port manually, but it worked when specifying only IP address

**Comment by paulreimer** [comment]
Is it correct that the new state of upstream `aoo` has changed the protocol in a way which is not backwards-compatible? i.e., there's no way to run an aooserver that would support new clients (either from a future version of Sonobus that uses the latest library code, or interop with custom clients that are directly using the latest aoo library version?)

And similarly, that Sonobus will likely never adopt the new library, since that would break compatibilty existing Sonobus clients?

**Comment by essej** [comment]
> Is it correct that the new state of upstream `aoo` has changed the protocol in a way which is not backwards-compatible? i.e., there's no way to run an aooserver that would support new clients (either from a future version of Sonobus that uses the latest library code, or interop with custom clients that are directly using the latest aoo library version?)
> 
> And similarly, that Sonobus will likely never adopt the new library, since that would break compatibilty existing Sonobus clients?

Correct, the new upstream AOO is not backwards compatible with the version SonoBus has been using. HOWEVER, the plan is that SonoBus 2.0 which will use the new one will not attempt to be backward compatible with SonoBus 1.x, and the default public connection server will run on a new port. People still running an older SB at that time will only be able to work with people also using the older version.

**Comment by paulreimer** [comment]
Gotcha / makes sense!

Will Sonobus 2.0 be a separate app e.g. in the iOS app store? i.e. so that people can still obtain Sonobus 1.x if they are trying to extend an existing deployment?

**Comment by essej** [comment]
Yes, it will be a separate app, the last 1.x will still be available, but probably no longer really maintained...
